File: app/src/main/java/com/template/project/exception/ServerException.java
package com.template.project.exception;

/**
 *  接口正常请求后 服务器返回的异常
 */
public class ServerException extends RuntimeException {

    private int code;

    public ServerException(String detailMessage) {
        super(detailMessage);
        this.code = 200;
    }

    public ServerException(String detailMessage, int code) {
        super(detailMessage);
        this.code = code;
    }

    public int getCode() {
        return code;
    }
}

File: app/src/main/java/com/tes/frezzmart/exception/ServerException.kt
package com.tes.frezzmart.exception

/**
 * 接口正常请求后 服务器返回的异常
 */
class ServerException : RuntimeException {
    var code: Int
        private set

    constructor(detailMessage: String?) : super(detailMessage) {
        code = 200
    }

    constructor(detailMessage: String?, code: Int) : super(detailMessage) {
        this.code = code
    }

}